portal/frontend/src/main/kotlin/com/sourceplusplus/portal/page/TracesPage.kt

```diff
@@ -52,4 +115,428 @@ class TracesPage(
             }
         }
     }
+
+    private fun setupUI() {
+        if (js("hideOverviewTab") as Boolean) {
+            jq("#overview_link").css("display", "none")
+            jq("#sidebar_overview_link").css("display", "none")
+        }
+
+        jq("#span_info_panel").css("display", "none")
+        jq("#latest_traces_header").addClass("active_sub_tab")
+            .removeClass("inactive_tab")
+        jq("#top_trace_table").css("display", "")
+        jq("#trace_stack_table").css("visibility", "hidden")
+        jq("#traces_span").css("display", "unset")
+        jq("#trace_stack_span").css("display", "none")
+        jq("#segment_id_span").css("display", "none")
+
+        jq("#trace_stack_header").addClass("inactive_tab")
+            .removeClass("active_sub_tab")
+            .css("visibility", "hidden")
+
+        jq("#span_info_header").addClass("inactive_tab")
+            .removeClass("active_sub_tab")
+            .css("visibility", "hidden")
+
+        jq("input[type='text']").on("click") {
+            jq(this).select()
+        }
+
+        window.setInterval(updateOccurredLabels(), 2000)
+    }
+
+    private fun clickedDisplayTraceStack(appUuid: String, artifactQualifiedName: String, globalTraceId: String) {
+        eb.send(
+            ClickedDisplayTraceStack,
+            json(
+                "portalUuid" to portalUuid,
+                "appUuid" to appUuid,
+                "artifactQualifiedName" to artifactQualifiedName,
+                "traceId" to globalTraceId
+            )
+        )
+    }
+
+    fun clickedDisplaySpanInfo(
+        appUuid: String,
+        rootArtifactQualifiedName: String,
+        traceId: String,
+        segmentId: String,
+        spanId: Int
+    ) {
+        eb.send(
+            ClickedDisplaySpanInfo,
+            json(
+                "portalUuid" to portalUuid,
+                "appUuid" to appUuid,
+                "artifactQualifiedName" to rootArtifactQualifiedName,
+                "traceId" to traceId,
+                "segmentId" to segmentId,
+                "spanId" to spanId
+            )
+        )
+    }
+
+    fun clickedBackToTraces() {
```

Comment:
You need to implement this method for it to work. See: https://github.com/sourceplusplus/SourceMarker/blob/327e56aa516ceca9947ca25858735bfab52e8458/portal/frontend/src/main/kotlin/com/sourceplusplus/portal/template/Navbar.kt#L47